### PR TITLE
KOGITO-8063: [SWF Editor] Disable cross-language autocompletion in YAML

### PR DESCRIPTION
**Pull request by fantonangeli**
Closes: https://issues.redhat.com/browse/KOGITO-8063

"Trying the autocompletion in the current version, with just this YAML: `functions: []`, there are bugs in the position of the completion and in the format of the snippet, because the value is written in JSON.

Please, disable the autocompletion only for the cross-language nodes."

Please note: that because the code completion is disabled in cross-language YAML, in this PR I also hide the "+Add function..." CodeLens, which would have nothing to suggest.

KOGITO-8063

### CODE REVIEW

File: packages/serverless-workflow-language-service/src/channel/types.ts [hunk 1]
```diff
@@ -53,4 +60,5 @@ export interface CodeCompletionStrategy {
   formatLabel(label: string, completionItemKind: CompletionItemKind): string;
   shouldComplete(args: ShouldCompleteArgs): boolean;
   getStartNodeValuePosition(document: TextDocument, node: SwfLsNode): Position | undefined;
+  shouldCreateCodelens(_args: ShouldCreateCodelensArgs): boolean;
```
Comment (caponetto) [inline packages/serverless-workflow-language-service/src/channel/types.ts]:
```suggestion
  shouldCreateCodelens(args: ShouldCreateCodelensArgs): boolean;
```

File: packages/serverless-workflow-language-service/src/channel/getNodeFormat.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+ * limitations under the License.
+ */
+
+import * as jsonc from "jsonc-parser";
+import { load } from "yaml-language-server-parser";
+import { SwfLsNode } from "./types";
+import { FileLanguage } from "../api";
+
+/**
+ * Detected the format of a node's content.
```
Comment (caponetto) [inline packages/serverless-workflow-language-service/src/channel/getNodeFormat.ts]:
```suggestion
 * Detect the format of a node's content.
```

File: packages/serverless-workflow-language-service/src/channel/getNodeFormat.ts [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+ * Note: for strings in double quotes and boolean values it's not able to distinguish between JSON and YAML. JSON will be returned in those cases.
+ *
+ * @param content the content
+ * @param node the node
+ * @returns the FileLanguage, null if unrecognized
```
Comment (caponetto) [inline packages/serverless-workflow-language-service/src/channel/getNodeFormat.ts]:
```suggestion
 * @returns the FileLanguage, undefined if unrecognized
```

File: packages/serverless-workflow-language-service/tests/getNodeFormat.test.ts [hunk 4]
```diff
@@ -0,0 +1,256 @@
...
+      });
+
+      test("single line declaration / with one attribute / with content before and after", async () => {
+        expect(
+          getNodeFormatTester({
+            content: `{
+            "name": "GreetInPortuguese",
+            "data": {🎯
+              "language": "Portuguese",
+              "message": "Hello"
```
Comment (caponetto) [inline packages/serverless-workflow-language-service/tests/getNodeFormat.test.ts]:
```suggestion
              "message": "Olá"
```
😛

File: packages/serverless-workflow-language-service/tests/getNodeFormat.test.ts [hunk 5]
```diff
@@ -0,0 +1,256 @@
...
+            🎯  language: Portuguese
+              message: Hello
+            transition: GetGreeting `,
+            ls,
+          })
+        ).toBe(FileLanguage.YAML);
+      });
+    });
+  });
+});
```
Comment (caponetto) [inline packages/serverless-workflow-language-service/tests/getNodeFormat.test.ts:263]: Can we have some tests with both YAML and JSON content mixed together?
After all, this is the case we are avoiding.

**Review by caponetto — COMMENTED** [review]
Thanks, @fantonangeli. Left some minor comments inline.

**Review by caponetto — APPROVED** [review]

**Review by paulovmr — APPROVED** [review]

**Pull request merged**